src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -262,6 +265,15 @@ def shutdown(self):
         if driver_result != mesos_pb2.DRIVER_STOPPED:
             raise RuntimeError("Mesos driver failed with %i", driver_result)
 
+    def setEnv(self,name, value=None):
+        """
+        Sets environment variables into dictionary for each job.
+        """
+        if value is None and name not in os.environ:
+            raise RuntimeError("%s does not exist in current environment", name)
+
+        self.environment[name] = value or os.environ[name]
```

Comment:
The way this method is implemented it does two lookups into the os.environ dictionary. This can introduce race conditions and is not optimal. It also checks `value` twice, but differently, causing this method to fail in certain cases if the value is "". Rewrite this method to only use one lookup and one conditional on value. Hint: use a try-except statement. The last line of the method will read `self.environment[name] = value`